docs/design/2021-04-26-lock-view.md

```diff
@@ -0,0 +1,191 @@
+# TiDB Design Documents
+
+- Author(s): [longfangsong](https://github.com/longfangsong), [MyonKeminta](http://github.com/MyonKeminta)
+- Last updated: April 29, 2021
+- Discussion PR: N/A
+- Tracking Issue: https://github.com/pingcap/tidb/issues/24199
+
+## Table of Contents
+
+* [Introduction](#introduction)
+* [Motivation or Background](#motivation-or-background)
+* [Detailed Design](#detailed-design)
+* [Test Design](#test-design)
+    * [Functional Tests](#functional-tests)
+    * [Scenario Tests](#scenario-tests)
+    * [Compatibility Tests](#compatibility-tests)
+    * [Benchmark Tests](#benchmark-tests)
+* [Impacts & Risks](#impacts--risks)
+* [Investigation & Alternatives](#investigation--alternatives)
+* [Unresolved Questions](#unresolved-questions)
+
+## Introduction
+
+This document describes the design of the feature Lock View, which provides tools to analyze problems about transaction's lock waiting, lock contentions and deadlocks.
+
+## Motivation or Background
+
+Currently, it's very hard to analyze lock contentions and deadlocks for transactions. One may need to enable general log, try to reproduce the problem, and try to analyze the log to find the cause, which is very difficult and inconvenient. We also found that this way of analyzing is not feasible in some scenarios. It's highly required to provide some better approach to analyze these kinds of problems.
+
+## Detailed Design
+
+Several tables will be provided in `information_schema`. Some tables has both local version (fetches data on the current TiDB node) and global version (fetches data among the whole cluster), and the global version's table name has the `"CLUSTER_"` prefix.
+
+### Table `(CLUSTER_)TIDB_TRX`
+
+| Field      | Type       | Comment |
+|------------|------------|---------|
+|`TRX_ID`    | `unsigned bigint`   | The transaction ID (aka. start ts) |
+|`TRX_STARTED`|`time`| Human readable start time of the transaction |
+|`DIGEST`|`text`| The digest of the current executing SQL statement |
+|`SQLS` | `text` | A list of all executed SQL statements' digests |
+|`STATE`| `enum('Running', 'Lock waiting', 'Committing', 'RollingBack')`| The state of the transaction |
+| `WAITING_START_TIME` | `time` | The elapsed time since the start of the current lock waiting (if any) |
+| `SCOPE` | `enum('Global', 'Local')` | The scope of the transaction |
+| `ISOLATION_LEVEL` | `enum('RR', 'RC')` | |
```

Comment:
Maybe it's better to use `REPEATABLE-READ` and `READ-COMMITTED` instead of the abbreviations, to keep the format consistent with elsewhere, e. g. the variable `@@tx_isolation` .